### ISSUE TITLE
Support for .NET MAUI

### ISSUE THREAD

**Issue by sattasundar** [roubachof/Sharpnado.Tabs#56]
As a developer, this plugin is very useful for customizing tab to meet the requirement. 

So, If this is going to support MAUI, it will be very useful.

Thanks.

**Comment by softlion** [comment]
I have the same question. I'm migrating an app to maui which uses the TabHostView control. Would be great to know if it's something that won't happen or will happen and a deadline. Ty !

**Comment by sattasundar** [comment]
> yup, but I wait for MAUI to be stable cause for now, it's a waste of time.

@roubachof  now, i think it's time to upgrade to MAUI.

**Comment by ewerspej** [comment]
> > yup, but I wait for MAUI to be stable cause for now, it's a waste of time.
> 
> @roubachof now, i think it's time to upgrade to MAUI.

I disagree. Although I'm already using MAUI, it's still very buggy despite it being GA already. Also, it would make more sense to fork this and create a MAUI variant instead of migrating this to MAUI and leaving Xamarin.Forms behind entirely, because XF will still live on for a while until a majority made the switch.

**Comment by roubachof** [comment]
Hi there !
In fact I'm using a strategy where I link XF files and using compilation directive to share the code between the 2 frameworks.
XF won't be left behind...

![Screenshot_1664374756](https://user-images.githubusercontent.com/596903/192803705-e0d75ba7-edca-4167-b934-845de5f7387d.png)

And also I'm currently working on it, sneak peak for this issue x)

**Comment by ewerspej** [comment]
@roubachof Is this compatible with Shell navigation?

**Comment by roubachof** [comment]
nope I don't support shell, you can't achieve advanced scenario with it.
I never used it in prod, Prism is way much better

**Comment by ewerspej** [comment]
> nope I don't support shell, you can't achieve advanced scenario with it.
> I never used it in prod, Prism is way much better

That's a pity, because Prism is overkill for small projects. For larger apps, I understand. Did you have a look at MAUI's new Shell?

**Comment by softlion** [comment]
> That's a pity, because Prism is overkill for small projects. For larger apps, I understand. Did you have a look at MAUI's new Shell?

I agree Shell has limitations and "bugs". For ex views in tabs are never released once they are created. But it's out of the box and will be used by a majority of users.
Prism has other limitations, does not play well with AOT, is fat, and mobile is not a first class citizen (ie: bugs/crashes arise with new versions).

**Comment by ewerspej** [comment]
@softlion I don't think it's a bug that views are not released in Shell, it seems more like a design decision. Those views are created lazily and since the views that are behind the tabs are frequently used views, this is not an issue, I think.

**Comment by roubachof** [comment]
If a dev is happy with the customization of Shell tabs, he should definitely use them. 
The philosophy of Tabs is to have full control of your views.
You can put them really where you want on your page and implement the design you want.

**Comment by softlion** [comment]
> it seems more like a design decision. 
I agree

> this is not an issue, I think.
i don't agree. They are created only once, reusing the same viewmodel (never disposing it).
If that was explained somewhere - as it's a major change from previous versions - ok. But it's hidden deep and official devs only beat around the bush.

> I didn't mean to divert the discussion
ok

> I was just curious whether it makes sense to plan with Sharpnado.Tabs for Shell apps or not, because the native Tabs are very limited in design and functionality. I will probably check out the TabView control that is being developed for the MAUI CommunityToolkit as well.
I agree.

**Comment by ewerspej** [comment]
@softlion Do you have some example code or a repo to check your solution out?

**Comment by roubachof** [comment]
Would it makes sense to make a specific tabs which integrates into shell? I have no idea since I never had an interest in shell ¯\_(ツ)_/¯

**Comment by ewerspej** [comment]
> Would it makes sense to make a specific sharpnado tabs version which integrates into shell? I have no idea since I never had an interest in shell ¯_(ツ)_/¯

I like the routing and the built-in (automatic) dependency injection of Shell in MAUI. So, I think it's certainly something to think about. It could be a joint effort, maybe?

**Comment by roubachof** [comment]
Please open a new issue for Shell integration cause Sharpnado.Tabs.Maui has now been shipped \o/

**Comment by NPadrutt** [comment]
@roubachof just for my understanding. Can I use Sharpnado to make a single page with tabs within an application that uses appshell or does that not work either?

**Comment by roubachof** [comment]
You can put tabs and the view switcher absolutely anywhere in your page, they are just very modular normal views
